File: app/src/main/java/com/zane/smapiinstaller/logic/ConfigManager.java
package com.zane.smapiinstaller.logic;

import android.os.Environment;

import com.zane.smapiinstaller.constant.Constants;
import com.zane.smapiinstaller.entity.FrameworkConfig;
import com.zane.smapiinstaller.utils.FileUtils;

import java.io.File;

/**
 * 配置管理器
 * @author Zane
 */
public class ConfigManager {
    private FrameworkConfig config;

    public ConfigManager() {
        File configFile = new File(Environment.getExternalStorageDirectory(), Constants.CONFIG_PATH);
        if(configFile.exists()) {
            config = FileUtils.getFileJson(configFile, FrameworkConfig.class);
        }
        if(config == null) {
            config = new FrameworkConfig();
            config.setInitial(true);
        }
    }

    public FrameworkConfig getConfig() {
        return config;
    }

    public void flushConfig() {
        File configFile = new File(Environment.getExternalStorageDirectory(), Constants.CONFIG_PATH);
        FileUtils.writeFileJson(configFile, config);
    }
}

File: app/src/main/java/com/zane/smapiinstaller/logic/ConfigManager.kt
package com.zane.smapiinstaller.logic

import com.zane.smapiinstaller.constant.Constants
import com.zane.smapiinstaller.entity.FrameworkConfig
import com.zane.smapiinstaller.utils.FileUtils
import java.io.File

/**
 * 配置管理器
 * @author Zane
 */
class ConfigManager {
    var config: FrameworkConfig

    init {
        val configFile = File(FileUtils.stadewValleyBasePath, Constants.CONFIG_PATH)
        var conf: FrameworkConfig? = null
        if (configFile.exists()) {
            conf = FileUtils.getFileJson(configFile, FrameworkConfig::class.java)
        }
        if (conf != null) {
            config = conf
        }
        else {
            config = FrameworkConfig()
            config.setInitial(true)
        }
    }

    fun flushConfig() {
        val configFile = File(FileUtils.stadewValleyBasePath, Constants.CONFIG_PATH)
        FileUtils.writeFileJson(configFile, config)
    }
}